src/wallet/wallet.ts

```diff
@@ -229,6 +230,16 @@ export const generateLyraWallet = async (testnet: boolean, mnem: string): Promis
  */
  export const generateAdaWallet = async (mnemonic: string): Promise<Wallet> => {
     return { mnemonic, xpub: await cardano.generateXPublicKey(mnemonic) };
+ }
+
+ /**
+ * Generate Xtz wallet
+ * @param mnem mnemonic seed to use
+ * @returns wallet
+ */
+ export const generateXtzWallet = async (mnem: string): Promise<Wallet> => {
+    const seed = mnemonicToSeedSync(mnem);
+    return { xpub: seed.toString('hex'), mnemonic: mnem };
```

Comment:
This cannot be xpub. Because it contains private keys inside. This must be derived path.